agent/xds/delta_test.go

```diff
@@ -1095,13 +1097,15 @@ func TestServer_DeltaAggregatedResources_v3_ACLEnforcement(t *testing.T) {
 
 			// If there is no token, check that we increment the gauge
 			if tt.token == "" {
-				data := scenario.sink.Data()
-				require.Len(t, data, 1)
-
-				item := data[0]
-				val, ok := item.Gauges["consul.xds.test.xds.server.streamsUnauthenticated"]
-				require.True(t, ok)
-				require.Equal(t, float32(1), val.Value)
+				retry.Run(t, func(r *retry.R) {
+					data := scenario.sink.Data()
+					require.Len(r, data, 1)
+
+					item := data[0]
+					val, ok := item.Gauges["consul.xds.test.xds.server.streamsUnauthenticated"]
+					require.True(r, ok)
```

Comment:
This assertion was failing randomly; I'm not 100% confident this will fix the issue but since `scenario.sink.Data` is timing-based I'm hoping retries will help here.

Reply:
When reading this code earlier, I assumed this assertion was failing randomly because this key `consul.xds.test.xds.server.streamsUnauthenticated` isn't always found in the data from #L1101. 
- #L1102 validates that data exists and the value isn't empty; which leads me to assume that the key requested isn't found in #L1104
- Is there a precise set of data that we use for testing that will always return this key/value?

Reply:
The gauge should be incremented in a normal codepath that is taken during the test: [1](https://github.com/hashicorp/consul/blob/72a73661c95800c1e3641af5d88555b4b019b1c9/agent/xds/server.go#L136), [2](https://github.com/hashicorp/consul/blob/b3ddd4d24efd42cc10db34e99400f44c30d7ec37/agent/xds/delta.go#L50)

I think the tests could be flaking because either:
1. There is some timing issue with the in-memory metrics sink
2. There is a data race between goroutines within the test

I didn't spend too much time looking into it but added a bunch of fixes for possible culprits